strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/artifact/MavenArtifactUtils.java

```diff
@@ -1,76 +1,93 @@
     {
-        String path = RepositoryFiles.relativizePath(repositoryPath);
-        Artifact artifact = ArtifactUtils.convertPathToArtifact(path);
-        
-        return new MavenRepositoryArtifact(artifact, repositoryPath);
+        final Gav gav = new Gav(artifact.getGroupId(),
+                                StringUtils.defaultString(artifact.getArtifactId()),
+                                StringUtils.defaultString(artifact.getVersion()),
+                                artifact.getClassifier(),
+                                artifact.getType(),
+                                null,
+                                null,
+                                null, false,
+                                null,
+                                false,
+                                null);
+        return M2_GAV_CALCULATOR.gavToPath(gav).substring(1);
     }
 
-    public static String getSnapshotBaseVersion(String version)
+
+    public static MavenArtifact convertPathToArtifact(RepositoryPath repositoryPath)
+            throws IOException
     {
-        return getSnapshotBaseVersion(version, true);
+        final Gav gav = convertPathToGav(repositoryPath);
+        return gav != null ? new MavenRepositoryArtifact(gav, repositoryPath) : null;
     }
 
-    public static String getSnapshotBaseVersion(String version,
-                                                boolean appendSnapshotSuffix)
+    public static MavenArtifact convertPathToArtifact(String path)
     {
-        return ArtifactUtils.getSnapshotBaseVersion(version, appendSnapshotSuffix);
+        final Gav gav = convertPathToGav(path);
+        return gav != null ? new MavenRepositoryArtifact(gav) : null;
     }
 
-    public static String getArtifactLevelMetadataPath(Artifact artifact)
+    public static Gav convertPathToGav(RepositoryPath repositoryPath)
+            throws IOException
     {
-        return ArtifactUtils.getArtifactLevelMetadataPath(artifact);
+        final String path = RepositoryFiles.relativizePath(repositoryPath);
+        return convertPathToGav(path);
     }
 
-    public static MavenArtifact getPOMArtifact(MavenArtifact source)
+    public static Gav convertPathToGav(String path)
     {
-        Artifact artifact = ArtifactUtils.getPOMArtifact(source);
-        return new MavenRepositoryArtifact(artifact, source.getPath());
+        return M2_GAV_CALCULATOR.pathToGav(path);
     }
 
-    public static boolean isSnapshot(String version)
+    public static boolean isGAV(RepositoryPath repositoryPath)
     {
-        return ArtifactUtils.isSnapshot(version);
+        try
+        {
+            return convertPathToGav(repositoryPath) != null;
+        }
+        catch (Exception ex)
+        {
+            logger.warn(ex.getMessage(), ex);
+        }
+        return false;
     }
 
-    public static String getVersionLevelMetadataPath(MavenArtifact artifact)
+    public static Pair<String, String> getDirectoryGA(RepositoryPath directoryPath)
```

Comment:
@fuss86 it would be good to have method javadoc explaining with examples what this `Pair` instance consist of.

Reply:
ok, will do